Add/dot pager component for carousel testimonials

mavegaf opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Implement a carousel component for Jetpack, based on the [Calypso dot-pager](https://github.com/Automattic/wp-calypso/tree/trunk/packages/components/src/dot-pager).
* Use the dot-pager on the [onboarding screen](https://github.com/Automattic/jetpack/blob/ef96ef73295bd042b215c02de000c72b68a13a50/projects/packages/my-jetpack/_inc/components/onboarding-screen/index.tsx#L27) to showcase testimonials.
* Include sample testimonials for demonstration purposes.

**Notes:**
* ~~Autorotation and pause-on-hover will be handled in a separate PR.~~
* RTL support is marked as TODO.
* Some tests will be addressed in a different PR.

https://github.com/user-attachments/assets/ba096a9f-2848-4dd6-ad94-2340be58c0b2



### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Spin up a JN site with Jetpack Beta pointing to this branch.
* Navigate to /wp-admin/admin.php?page=my-jetpack#/onboarding.
* In the slider, verify that:
  * It slides when dragging on the screen.
  * It slides when clicking the dots.
  * It adjusts correctly to different screen sizes.

mavegaf committed:
Copying and working on initial files for carusel testimonials using dot-pager

mavegaf committed:
Adding styles

mavegaf committed:
fix swipeable types errors

mavegaf committed:
Adding scss (commented)

mavegaf committed:
changelog

mavegaf committed:
add some styles

mavegaf committed:
fix some styles

mavegaf committed:
Moving dots to bottom

mavegaf committed:
fix fit components

mavegaf committed:
remove unnecesary di

mavegaf committed:
Remove unused z-index

mavegaf committed:
re-enable @include reduce-motion in scss

mavegaf committed:
Add some testimonials for testing purposes

mavegaf committed:
changelog

mavegaf committed:
fix style path

mavegaf committed:
changing images

mavegaf committed:
Adding some README and tests

mavegaf committed:
fix import order

mavegaf committed:
adjust background position center top

mavegaf committed:
other css changes

mavegaf committed:
organize dot-page and css

mavegaf committed:
Improve text align on size change

mavegaf committed:
remove unused tests

mavegaf committed:
Configure autoswipe and stop on hover

mavegaf commented:
Added configuration for the autorotation and added pause-on-hover since it was a small change

grzegorz-cp commented on projects/packages/my-jetpack/_inc/components/testimonials/index.tsx:
```diff
@@ -0,0 +1,46 @@
+			<DotPager rotateTime={ 5 }>
+				<div className="testimonial1">
+					<div className="testimonial-content">
+						<p className="header">Grow your audience with Jetpack.</p>
```
Please wrap the strings in the translation method.

mavegaf replied:
👍

grzegorz-cp commented on projects/packages/my-jetpack/_inc/components/testimonials/index.tsx:
```diff
@@ -0,0 +1,46 @@
+					<div className="testimonial-content">
+						<p className="header">Grow your audience with Jetpack.</p>
+						<p className="header">Speed up your site</p>
+						<p className="header">Keep it secure.</p>
```
Those paragraphs are only for the first testimonial, but it has three headers and the other two have none.

Can you update the class names and apply the BEM convention to the component?

mavegaf replied:
Done! CSS using BEM

grzegorz-cp commented on projects/packages/my-jetpack/_inc/components/testimonials/style.scss:
```diff
@@ -0,0 +1,75 @@
+    background-size: cover;
+    background-position: center top;
+    background-repeat: no-repeat;
+    font-family: 'Open Sans', sans-serif;
```
Please use SCSS variables from Jetpack that already have font names and colours defined.

Are all those CSS rules needed to achieve the desired effect?

grzegorz-cp commented on projects/packages/my-jetpack/_inc/components/testimonials/style.scss:
```diff
@@ -0,0 +1,75 @@
+.testimonial3 {
+    @extend .testimonial-base;
+    background-image: url('./tim-ferriss-testimonial.png');
+    background-color: #3f3f3f;
```
Please avoid hard-coding the colours. Do all of the testimonials need a different background colour?

mavegaf replied:
Oh no. this background-color was a test I missed to delete

mavegaf replied:
Done! Removed and using colors from Jetpack css variables

grzegorz-cp commented on projects/js-packages/components/components/dot-pager/style.scss:
```diff
@@ -0,0 +1,86 @@
+@import '@automattic/jetpack-base-styles/gutenberg-base-styles';
+
+$studio-gray-10: #c3c4c7;
```
Can you take the colours from the global Jetpack CSS variables? You can also provide a fallback value if they are not defined.

mavegaf replied:
Done!

grzegorz-cp reviewed (changes requested):
Thank you for creating the components. I left a few comments.
This is the right direction, but we need to revisit CSS and a few other things, such as translations.

Unfortunately, the text placement and size are not correct on the slides. I'll send you a link to the designs on Slack.

Currently, the mobile version of the page hides the testimonials, but it should be on top of the page according to the designs. If this is too much, we can address the mobile version in another PR.

mavegaf committed:
Use css BEM and set some colors and fonts

mavegaf committed:
adding translations

mavegaf committed:
Using jetpack variable colors on dot-pager css

mavegaf committed:
Using jetpack variable colors on testimonials css

mavegaf committed:
Using Jetpack variables for font size. Removing for now font-family

mavegaf commented:
@grzegorz-cp Thanks for the review. I have addressed most of the comments:
* CSS using BEM
* Adding translations
* Using Jetpack variables for colors

I'll leave some questions about the fonts and the design in Slack

grzegorz-cp commented on projects/packages/my-jetpack/_inc/components/testimonials/index.tsx:
```diff
@@ -0,0 +1,62 @@
+	return (
+		<>
+			<DotPager rotateTime={ 5 }>
+				<div className="testimonial testimonial--1">
```
In a follow-up PR it would be good to replace generic `testimonial` with something like `jetpack-onboarding-testimonial` because we risk CSS bleeding with generic names.

grzegorz-cp reviewed (approved):
Thank you for making the changes. This looks good.
I left a suggestion for a follow-up PR regarding CSS class naming.

The main slide text should have more spacing between the text and the dot pager.
Also, font sizes are too small compared to the design. The quote text should be 24px, name 20px and position 16px. This is an improvement, so please ship without it and update in the next PR.
